core/src/main/java/org/infinispan/remoting/rpc/RpcManagerImpl.java

```diff
@@ -85,8 +98,26 @@ private void start() {
 
       if (configuration.transaction().transactionProtocol().isTotalOrder())
          t.checkTotalOrderSupported();
+
+      configuration.clustering()
+                   .attributes().attribute(ClusteringConfiguration.REMOTE_TIMEOUT)
+                   .addListener(updateRpcOptions);
+      updateRpcOptions(configuration.clustering().attributes().attribute(ClusteringConfiguration.REMOTE_TIMEOUT), null);
+   }
+
+   @Stop
+   private void stop() {
+      configuration.clustering()
+                   .attributes().attribute(ClusteringConfiguration.REMOTE_TIMEOUT)
+                   .removeListener(updateRpcOptions);
+   }
+
+   private void updateRpcOptions(Attribute<Long> attribute, Long oldValue) {
+      syncRpcOptions = new RpcOptions(DeliverOrder.NONE, attribute.get(), TimeUnit.MILLISECONDS);
+      totalSyncRpcOptions = new RpcOptions(DeliverOrder.TOTAL, attribute.get(), TimeUnit.MILLISECONDS);
```

Comment:
I think not, but I wanted the name to say that it's sync so I didn't have to check :)